azext_edge/edge/providers/rpsaas/base_provider.py

```diff
@@ -0,0 +1,87 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from typing import Any, Dict, List, Optional
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class RPSaaSBaseProvider:
+    def __init__(
+        self,
+        cmd,
+        api_version: str,
+        resource_type: str,
+        **kwargs
+    ):
+        from azure.cli.core.commands.client_factory import get_subscription_id
+        from ...util.az_client import get_resource_client
+
+        self.cmd = cmd
+        self.subscription = get_subscription_id(cmd.cli_ctx)
+        self.resource_client = get_resource_client(subscription_id=self.subscription)
+        self.api_version = api_version
+        self.resource_type = resource_type
+        self.required_extension = kwargs.get("required_extension")
+
+    def delete(
+        self,
+        resource_name: str,
+        resource_group_name: str
+    ):
+        return self.resource_client.resources.begin_delete(
+            resource_group_name=resource_group_name,
+            resource_provider_namespace=self.resource_type,
+            parent_resource_path="",
+            resource_type="",
+            resource_name=resource_name,
+            api_version=self.api_version
+        )
+
+    def get_location(
+        self,
+        resource_group_name: str
+    ) -> str:
+        resource_group = self.resource_client.resource_groups.get(resource_group_name=resource_group_name)
+        return resource_group.as_dict()["location"]
+
+    def list(
+        self,
+        resource_group_name: Optional[str] = None,
+    ) -> List[Any]:
+        # Note the usage of az rest/send_raw_request over resource
+        # az resource list/resource_client.resources.list will omit properties
+        from ...util.common import _process_raw_request
+        uri = f"/subscriptions/{self.subscription}"
+        if resource_group_name:
+            uri += f"/resourceGroups/{resource_group_name}"
+        uri += f"/providers/{self.resource_type}?api-version={self.api_version}"
+        return _process_raw_request(
+            cmd=self.cmd, method="GET", url=uri, keyword="value"
+        )
+
+    def show(
+        self,
+        resource_name: str,
+        resource_group_name: str
+    ) -> Dict[str, Any]:
+        result = self.resource_client.resources.get(
+            resource_group_name=resource_group_name,
+            resource_provider_namespace=self.resource_type,
```

Comment:
Interesting that resource_type value is used for resource_provider_namespace

Reply:
Here is where you can get the definitions for everything (it corresponds to the params directly)
![image](https://github.com/Azure/azure-iot-ops-cli-extension/assets/73560279/72d2b572-284b-417e-98a0-4dd644677346)
So by these definitions, assets would have
resource_namespace = Microsoft.DeviceRegistry
resource_type = assets

The nice thing is that if you combine them (like I did above in namespace) - it still works correctly. You just need to set the others to ""

Since I like to be lazy and reduce number of enums to track and match - I combined them and put them into namespace. Testing - it works if you put them into resource_type as so:
![image](https://github.com/Azure/azure-iot-ops-cli-extension/assets/73560279/5fee33f4-f8f7-4025-9ff3-2e826170f84b)

So to follow the spirit of the `--resource-type` param help - I will just move the arg from resource_provider_namespace to resource_type